src/gwt/src/org/rstudio/studio/client/common/console/ConsoleProcess.java

```diff
@@ -254,7 +254,64 @@ public ConsoleProgressDialog showConsoleProgressDialog(
          
          return dlg;
       }
+      
+      /**
+       * Add process to list of processes, sorted in ascending order by
+       * terminal sequence number. If duplicate sequence numbers are
+       * encountered, all but the first will have the process killed.
+       * 
+       * @param terminalProcs (in/out) sorted list of terminal processes
+       * @param cproc process to insert in the list
+       */
+      private static void addTerminalProc(ArrayList<ConsoleProcess> terminalProcs,
+                                          ConsoleProcess cproc)
+      {
+         int newSequence = cproc.getProcessInfo().getTerminalSequence();
+
+         if (newSequence < 1)
+         {
+            Debug.log("Invalid terminal sequence, killing unrecognized process");
```

Comment:
Use `Debug.logWarning` for this kind of thing as it also prints a stack trace -- we very rarely use `Debug.log` for anything other than printf-style debugging. (Also consider logging the observed sequence number and anything in the process info that might aid diagnosis)